src/views/index-dtf/issuance/async-swaps/async-mint/index.tsx

```diff
@@ -0,0 +1,152 @@
+import {
+  asyncSwapFetchingAtom,
+  asyncSwapInputAtom,
+  asyncSwapOngoingTxAtom,
+  asyncSwapRefetchAtom,
+  asyncSwapResponseAtom,
+  bufferValueAtom,
+  collateralAcquiredAtom,
+  isMintingAtom,
+  mintValueAtom,
+  mintValueUSDAtom,
+  mintValueWeiAtom,
+  selectedTokenBalanceAtom,
+  selectedTokenOrDefaultAtom,
+  slippageAtom,
+} from '../atom'
+import CollateralAcquisition from '../collateral-acquisition'
+import SubmitMint from './submit-mint-orders'
+
+const AsyncMint = () => {
+  const chainId = useAtomValue(chainIdAtom)
+  const indexDTF = useAtomValue(indexDTFAtom)
+  const [inputAmount, setInputAmount] = useAtom(asyncSwapInputAtom)
+  const selectedToken = useAtomValue(selectedTokenOrDefaultAtom)
+  const selectedTokenBalance = useAtomValue(selectedTokenBalanceAtom)
+  const isMinting = useAtomValue(isMintingAtom)
+  const slippage = useAtomValue(slippageAtom)
+  const [ongoingTx, setOngoingTx] = useAtom(asyncSwapOngoingTxAtom)
+  const setAsyncSwapRefetch = useSetAtom(asyncSwapRefetchAtom)
+  const setAsyncSwapFetching = useSetAtom(asyncSwapFetchingAtom)
+  const selectedTokenPrice = useChainlinkPrice(chainId, selectedToken.address)
+  const inputPrice = (selectedTokenPrice || 0) * Number(inputAmount)
+  const onMax = () => setInputAmount(selectedTokenBalance?.balance || '0')
+  const asyncSwapResponse = useAtomValue(asyncSwapResponseAtom)
+  const collateralAcquired = useAtomValue(collateralAcquiredAtom)
+  const orderSubmitted = !!asyncSwapResponse
+  const amountOut = useAtomValue(mintValueAtom)
+  const amountOutWei = useAtomValue(mintValueWeiAtom)
+  const amountOutValue = useAtomValue(mintValueUSDAtom)
+  const bufferValue = useAtomValue(bufferValueAtom)
+
+  const insufficientBalance =
+    parseUnits(inputAmount, selectedToken.decimals) >
+    (selectedTokenBalance?.value || 0n)
+
+  const { data, isLoading, isFetching, refetch, failureReason } = useAsyncSwap({
+    dtf: indexDTF?.id,
+    amountOut: amountOutWei.toString(),
+    slippage: isFinite(Number(slippage)) ? Number(slippage) : 10000,
+    disabled: insufficientBalance || ongoingTx,
+    dtfTicker: indexDTF?.token.symbol || '',
+    type: 'mint',
+  })
+
+  const { loadingAfterRefetch } = useLoadingAfterRefetch(data)
+
+  const priceFrom = 0
+
+  // const valueTo = data?.result?.amountOut
+  // const showTxButton = Boolean(
+  //   data?.status === 'success' &&
+  //     data?.result &&
+  //     !insufficientBalance &&
+  //     !isLoading
+  // )
+  const awaitingQuote = isLoading || isFetching
+
+  useEffect(() => {
+    setAsyncSwapRefetch({ fn: refetch })
+  }, [refetch, setAsyncSwapRefetch])
+
+  useEffect(() => {
+    setAsyncSwapFetching(awaitingQuote)
+  }, [awaitingQuote, setAsyncSwapFetching])
+
+  useEffect(() => {
+    setOngoingTx(false)
+    setInputAmount('')
+  }, [])
+
+  if (!indexDTF) return null
+
+  return (
+    <div className="flex flex-col h-full">
+      <Swap
```

Comment:
Replace the "Swap" component for a component tree on which each component talk with the specific state they care about, current component is too big